dpctl/_sycl_platform.pyx

```diff
@@ -0,0 +1,289 @@
+            filter_c_str = string
+            DSRef = DPCTLFilterSelector_Create(filter_c_str)
+            ret = self._init_from_selector(DSRef)
+            if ret == -1:
+                raise ValueError(
+                    "Could not create a SyclPlatform with the selector string"
+                )
+            # Free up the device selector
+            DPCTLDeviceSelector_Delete(DSRef)
+        elif isinstance(arg, _SyclPlatform):
+            ret = self._init_from__SyclPlatform(arg)
+            if ret == -1:
+                raise ValueError(
+                    "Could not create SyclPlatform from _SyclPlatform instance"
+                )
+        elif arg is None:
+            PRef = DPCTLPlatform_Create()
+            if PRef is NULL:
+                raise ValueError(
+                    "Could not create a SyclPlatform from default selector"
+                )
+            else:
+                SyclPlatform._init_helper(self, PRef)
+        else:
+            raise ValueError(
+                "Invalid argument. Argument should be a str object specifying "
+                "a SYCL filter selector string."
+            )
+
+    def print_platform_info(self):
+        """ Print information about the SYCL platform.
+        """
+        DPCTLPlatformMgr_PrintInfo(self._platform_ref)
+
+    @property
+    def vendor(self):
+        """ Returns the platform vendor name as a string.
+        """
+        return self._vendor.decode()
+
+    @property
+    def version(self):
+        """ Returns a backend-defined driver version as a string.
+        """
+        return self._version.decode()
+
+    @property
+    def name(self):
+        """ Returns the name of the platform as a string
+        """
+        return self._name.decode()
+
+    @property
+    def backend(self):
+        """Returns the backend_type enum value for this device
+
+        Returns:
+            backend_type: The backend for the device.
+        """
+        cdef DPCTLSyclBackendType BTy = (
+            DPCTLPlatform_GetBackend(self._platform_ref)
+        )
+        if BTy == _backend_type._CUDA:
+            return backend_type.cuda
+        elif BTy == _backend_type._HOST:
+            return backend_type.host
+        elif BTy == _backend_type._LEVEL_ZERO:
+            return backend_type.level_zero
+        elif BTy == _backend_type._OPENCL:
+            return backend_type.opencl
+        else:
+            raise ValueError("Unknown backend type.")
+
+
+def lsplatform():
+    """
+    Prints out the list of available SYCL platforms and various information
+    about each platform.
+
+    Currently, this function prints a list of all SYCL platforms that
+    are available on the system and the list of devices for each platform.
+
+    :Example:
+        On a system with an OpenCL CPU driver, OpenCL GPU driver,
+        Level Zero GPU driver, running the command. ::
+
+        $python -c "import dpctl; dpctl.print_info()"
```

Comment:
Should this docstring reflect the actual name of the function, i.e. replace `dpctl.print_info()` with `dpctl.lsplatform()` ?

Reply:
done.